crates/aptos/src/governance/mod.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::common::types::{CliTypedResult, TransactionOptions};
+use crate::{CliCommand, CliResult};
+use aptos_rest_client::Transaction;
+use aptos_types::account_address::AccountAddress;
+use async_trait::async_trait;
+use clap::Parser;
+
+/// Tool for on-chain governance
+///
+#[derive(Parser)]
+pub enum GovernanceTool {
+    Propose(SubmitProposal),
+    Vote(SubmitVote),
+}
+
+impl GovernanceTool {
+    pub async fn execute(self) -> CliResult {
+        use GovernanceTool::*;
+        match self {
+            Propose(tool) => tool.execute_serialized().await,
+            Vote(tool) => tool.execute_serialized().await,
+        }
+    }
+}
+
+/// Submit proposal to other validators to be proposed on
+#[derive(Parser)]
+pub struct SubmitProposal {
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+    /// Delegated pool address to submit proposal on behalf of
+    #[clap(long)]
+    pub(crate) pool_address: AccountAddress,
+    /// Execution hash of the script to be voted on
+    #[clap(long)]
+    pub(crate) execution_hash: String,
+    /// Code location of the script to be voted on
+    #[clap(long)]
+    pub(crate) code_location: String,
+    /// Title of proposal
+    #[clap(long)]
+    pub(crate) title: String,
+    /// Description of proposal
+    #[clap(long)]
+    pub(crate) description: String,
```

Comment:
Just curious is String easily serializable into bytes required for Move's utf8 string? We recently switched from ascii to utf8 on the Move side.

Reply:
Strings in Rust are always valid UTF-8, so this should be fine. Pretty sure they have an `as_bytes` method to convert them to bytes too.